### PR TITLE
Adding DTL VM facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Adding facts for DTL Virtual Machine.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_devtestlabvirtualmachine

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+            - The name of the resource group.
+        required: True
+    lab_name:
+        description:
+            - The name of the lab.
+        required: True
+    name:
+        description:
+            - The name of the virtual machine.
+    tags:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py:36]: Xiang mentioned that DTL tags is hiding, could you pls figure out how it works regarding adding tags and filter by tags

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+            returned: always
+            type: str
+            sample: myLab
+        name:
+            description:
+                - Name of the virtual machine.
+            returned: always
+            type: str
+            sample: myVm
+
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py]: redundant empty lines, not consistent formattting

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.devtestlabs import DevTestLabsClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMVirtualMachineFacts(AzureRMModuleBase):
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py]: wrong class name

File: test/integration/targets/azure_rm_devtestlab/tasks/main.yml [hunk 4]
```diff
@@ -315,6 +315,59 @@
...
+      - output.virtualmachines[0]['image'] != None
+      - output.virtualmachines[0]['notes'] != None
+      - output.virtualmachines[0]['os_type'] != None
+      - output.virtualmachines[0]['provisioning_state'] != None
+      - output.virtualmachines[0]['storage_type'] != None
+      - output.virtualmachines[0]['user_name'] != None
+      - output.virtualmachines[0]['vm_size'] != None
+
+- name: List Facts of DTL Virtual Machine
+  azure_rm_devtestlabvirtualmachine_facts:
```
Comment (yungezz) [inline test/integration/targets/azure_rm_devtestlab/tasks/main.yml:348]: adding filter by tags

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py:36]: I just rechecked that, and:
- when dtl vm is created tags are properly assigned to that dtl vm
- dtl  vm has "compute_id" property which is the id of real underlying vm
- real vm + additional resources are created in that separate resource group

so this fact implementation is correct and provides enough information to locate actual vm.

Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py:36]: so filtering by tag  on DTL vm is same as normal VM?

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_devtestlabvirtualmachine_facts.py:36]: yes

**Review by yungezz — APPROVED** [review]

**Pull request merged**